src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -841,7 +841,7 @@ def build_messages(
     image_detail: str = "auto",
     **kwargs,
 ):
-    inputs_to_escape = kwargs.pop(INPUTS_TO_ESCAPE_PARAM_KEY, None)
+    inputs_to_escape = kwargs.pop(INPUTS_TO_ESCAPE_PARAM_KEY, list(kwargs.keys()))
```

Comment:
This handle can work for 'prompty alone (prompty not in flow)' situation.

But for long term, it might not be sufficient for the scenario in flow, neither for prompty nor LLM node.

Under flow scenario, since if a node, either prompty or LLM, if it has no flow inputs, then executor would pass no 'INPUTS_TO_ESCAPE_PARAM_KEY' list. So with your change, all the node inputs would go through escape process, even a node input linked to a prompt tool output, which is a string. But this is not expected. The prompt tool string should get the roles parsed.

So in summary, this change works only for prompty alone sccenario. Please add a comment and fix this in following PRs to include the flow scenarios.

Reply:
add comment and not support prompty in flow currently